core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -915,6 +915,17 @@ void *cuda_resize_scratch_space(std::int64_t bytes, bool force_shrink) {
   return ptr;
 }
 
+void cuda_prefetch_pointer(const Cuda &space, void *const ptr, size_t bytes,
+                           bool to_device) {
+  cudaPointerAttributes attr;
+  cudaPointerGetAttributes(&attr, ptr);
+  int direction = to_device ? space.cuda_device() : cudaCpuDeviceId;
```

Comment:
Not used.

Reply:
ah ups. it always prefetches to the device right now … fixing it

Reply:
turns out prefetching towards the host is slowing things a bit down. so It now only prefetches in the device direction

Reply:
It would be helpful to document that observation in a short comment to avoid confusion why it doesn't do anything in the host direction.